asserts/asserts.go

```diff
@@ -158,16 +160,54 @@ func parseHeaders(head []byte) (map[string]string, error) {
 		return nil, fmt.Errorf("header is not utf8")
 	}
 	headers := make(map[string]string)
-	for _, entry := range strings.Split(string(head), "\n") {
-		nameValueSplit := strings.Index(entry, ": ")
+	lines := strings.Split(string(head), "\n")
+	for i := 0; i < len(lines); {
+		entry := lines[i]
+		i++
+		nameValueSplit := strings.Index(entry, ":")
 		if nameValueSplit == -1 {
-			return nil, fmt.Errorf("header entry missing name value ': ' separation: %q", entry)
+			return nil, fmt.Errorf("header entry missing name/value ':' separator: %q", entry)
```

Comment:
Much as I enjoy garden path sentences, I don't think they should be used for error messages. Perhaps a better option would be just

> header entry missing ':' separator

or

> header entry missing ':' separator between name and value

?